core/test.ts

```diff
@@ -24,8 +24,12 @@ export class Test {
     return this;
   }
 
-  public dataset(datasetToTest: string) {
-    this.datasetToTest = datasetToTest;
+  public dataset(ref: Resolvable) {
+    const target =
```

Comment:
let's not do this resolution here. instead, do it in compile(), when all actions must have been attached to the `session`. then look up the relevant `datasetToTest` in the session.

edit: in fact, you already do this! let's just remove this code; compile() does the right thing already. this can just be `this.datasetToTest = ref;`

Reply:
See my comments above about accepting `Resolvable` as an argument to `test.dataset()`. This was completely outside of the specs of the original issue so I am fine removing it.